fabric-rendering-v1/src/main/java/net/fabricmc/fabric/api/client/rendering/v1/BlockEntityRendererRegistry.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.client.rendering.v1;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import net.minecraft.block.entity.BlockEntity;
+import net.minecraft.client.render.block.entity.BlockEntityRenderDispatcher;
+import net.minecraft.client.render.block.entity.BlockEntityRenderer;
+
+/**
+ * Helper class for registering BlockEntityRenderers.
+ */
+public class BlockEntityRendererRegistry {
+	public static final BlockEntityRendererRegistry INSTANCE = new BlockEntityRendererRegistry();
+	private Map<Class<? extends BlockEntity>, BlockEntityRenderer<? extends BlockEntity>> renderers = null;
+	private Map<Class<? extends BlockEntity>, BlockEntityRenderer<? extends BlockEntity>> renderersTmp = new HashMap<>();
+
+	private BlockEntityRendererRegistry() { }
+
+	public void initialize(BlockEntityRenderDispatcher instance, Map<Class<? extends BlockEntity>, BlockEntityRenderer<? extends BlockEntity>> map) {
+		if (renderers != null && renderers != map) {
+			throw new RuntimeException("Tried to set renderers twice!");
```

Comment:
This should be `UnsupportedOperationException`
```suggestion
			throw new UnsupportedOperationException("Cannot set renderers more than once");
```

Reply:
I don’t actually know much about the existing rendering api, this is just copied from v1.

Reply:
just do the runtime exception. if you throw something else it should be illegal state @ChloeDawn

Reply:
Throwing unspecialized exceptions like RuntimeException is generally considered bad form. IllegalStateException may indeed be more fitting though.

Reply:
ISE is the right choice here, but better done outside this PR. Normally the module migration would be separate as well, but this is just testing the approach.